### PR TITLE
i#5302: Fix a32 assert on rel-addr opnd disassembly

### LINKED ISSUE #5302: ASSERT opnd_disassemble: invalid opnd type on rel-addr opnd on aarch32
Running a drbbdup test that creates a rel-addr opnd on aarch32:
```
SYSLOG_ERROR: Application suite/tests/bin/simple_app (2316) DynamoRIO usage error : opnd_disassemble: invalid opnd type
```

It looks like internal_opnd_disassemble() incorrectly has REL_ADDR_kind handling under ifdef X64 instead of X64 || ARM.

Xref #5295

### PR DESCRIPTION
**Pull request by derekbruening**
Adds correct ifdef ARM handling of REL_ADDR_kind operands when
disassembling, and in legacy loadtoconst code.

Adds a test to api.ir (confirmed it fails without this fix).

Fixes other latent issues in api.ir on arm which unfortunately does
not run on QEMU yet and so is not run automatically in today's CI.
Tested manually on an arm system.

Fixes #5302

### CODE REVIEW

File: suite/tests/api/ir_arm.c [hunk 1]
```diff
@@ -81,7 +81,10 @@ test_instr_encoding(void *dc, uint opcode, instr_t *instr)
     pc = instr_encode(dc, instr, buf);
```
Comment (abhinav92003) [inline suite/tests/api/ir_arm.c:81]: Assert pc != NULL? It was checked in the old version of `test_pcrel`.

File: suite/tests/api/ir_arm.c [hunk 2]
```diff
@@ -111,28 +114,26 @@ test_pred(void *dc)
...
 test_pcrel(void *dc)
 {
-    byte *pc;
     instr_t *inst;
     inst = INSTR_CREATE_ldr(dc, opnd_create_reg(DR_REG_R0),
                             opnd_create_rel_addr((void *)&buf[128], OPSZ_PTR));
-    pc = instr_encode(dc, inst, buf);
-    /* (gdb) x/2i buf+1
-     *    0x120b9 <buf+1>:     ldr     r0, [pc, #124]  ; (0x12138 <buf+128>)
+    /* On decoding our rel-addr opnd turns into a base-disp:
```
Comment (abhinav92003) [inline suite/tests/api/ir_arm.c:132]: Is it worth constructing an instruction with the expected base-disp operand and checking equality of `inst`  with that?

**Review by abhinav92003 — APPROVED** [review]

File: suite/tests/api/ir_arm.c (hunk 1 above)
Comment (derekbruening) [reply suite/tests/api/ir_arm.c:81]: Done.

File: suite/tests/api/ir_arm.c (hunk 2 above)
Comment (derekbruening) [reply suite/tests/api/ir_arm.c:132]: Duplicating the code from test_instr_encoding?  Maybe it's worth it to avoid the bool param.  OK, done.

Comment (abhinav92003) [reply suite/tests/api/ir_arm.c:131]: Not only to avoid the bool param. But also for a stronger test.

**Pull request merged**